### PR TITLE
migrate-performance-tips

### PR DESCRIPTION
**Pull request by nglillywhite**
Migrated the performance tips and tricks from fastai as per https://github.com/fastai/fastai/issues/3125
- Migrated performance.md
- Removed reference to check_perf() which hasn't been migrated but existed in fastai1
- Checked links & Legibility

### CODE REVIEW

**Comment by muellerzr** [comment]
@nglillywhite have you verified that all of it works? `fastai` no longer has a `pillow_version` function. You could do something like:

```python
import PIL
print(PIL.__version__)
```

(for instance right now, it will either say 7.0.0 or 7.0.0.post3, with the latter being simd)

**Comment by nglillywhite** [comment]
Thats a good point, let me verify all the commands and work through the doc. I read it for legibility and checked hyperlinks but didn't run the commands myself. I'll check them all and update accordingly, thanks!

**Comment by nglillywhite** [comment]
@muellerzr , you're right on the pillow version syntax. I updated the example to the latest available versions.

I'm on windows so I couldn't verify the path examples already provided in the other snippets but the rest of the installation commands worked pending having the prerequisites installed. Happy to spin up a VM and work through them if thats needed to get the PR through.

**Comment by muellerzr** [comment]
@jph00 I'd say this is ready for your review :) This would close https://github.com/fastai/fastai/issues/3125

File: docs_src/sidebar.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
     "Getting started": {
         "Overview": "/",
-        "Quick Start": "/quick_start.html"
+        "Quick Start": "/quick_start.html",
+        "Performance Tips and Tricks": "/dev/performance.html"
```
Comment (jph00) [inline docs_src/sidebar.json:5]: Please move this to the advanced tutorials sections.

**Review by jph00 — CHANGES_REQUESTED** [review]
Many thanks! Just one request :)

File: docs_src/sidebar.json (hunk 1 above)
Comment (nglillywhite) [reply docs_src/sidebar.json]: I've updated the branch but it still mentions changes required so I'm going to resolve this conversation in the hope that cleans it up. Apologies if I've just added more notifications unnecessarily.

**Pull request merged**